src/babel/transformation/file/options.json

```diff
@@ -21,8 +21,8 @@
   },
 
   "moduleId": {
-    "description": "specify a custom name for module ids",
-    "type": "string"
+    "description": "specify a custom name for module ids, or a function to transform the default name",
+    "type": ["string", "function"]
```

Comment:
This is the first option that allows overloading, is this correct and OK?

I prefer overloading in this case...since it's most-like other tools, but you're the boss..